### ISSUE TITLE
Public fields for derive(WrapperApi)

### ISSUE THREAD

**Issue by vurvdev** [szymonwieloch/rust-dlopen#31]
Why doesn't the macro allow the fields to be public? Or is there a way I don't know of?

I'm a noob at this stuff, so I don't know if there is a reason or if it just was never implemented. It's super frustrating that this is the only thing holding me back on my project..

**Comment by vurvdev** [comment]
Forgot to link where it's not allowed: https://github.com/szymonwieloch/rust-dlopen/blob/26b70292744db1052403378d4a01075b6171e3d1/rust-dlopen-derive/src/wrapper.rs#L17

**Comment by vurvdev** [comment]
It looks like public fields work fine on my fork. https://github.com/Vurv78/rust-dlopen/tree/future I also took the liberty of making dlopen-derive a feature so you don't need double extern crate / crate imports. Will make a pull request when I fix one thing about the macro exporting.

Edit: Seems I created more problems by changing the file structure, namely the tests and how they get the dll.

Edit 2: Found a reason on my own; https://github.com/rust-lang/rust-bindgen/issues/1937#issuecomment-737117665
